covenants/src/protocol_tests.rs

```diff
@@ -0,0 +1,437 @@
+                let client = client.clone();
+                |transaction| async move { client.sign_raw_transaction(&transaction).await }
+            })
+            .await
+            .unwrap();
+
+        let loan_transaction = lender
+            .finalise_loan(loan_transaction, {
+                let client = client.clone();
+                |transaction| async move { client.sign_raw_transaction(&transaction).await }
+            })
+            .await
+            .unwrap();
+
+        client
+            .send_raw_transaction(&loan_transaction)
+            .await
+            .unwrap();
+
+        client.generatetoaddress(1, &miner_address).await.unwrap();
+
+        let loan_repayment_transaction = borrower
+            .loan_repayment_transaction(
+                &mut thread_rng(),
+                &SECP256K1,
+                {
+                    let client = client.clone();
+                    |amount, asset| async move { find_inputs(&client, asset, amount).await }
+                },
+                {
+                    let client = client.clone();
+                    |mut tx, index, address, value| async move {
+                        let sk = client.dump_private_key(&address).await?;
+                        let pk = secp256k1_zkp::PublicKey::from_secret_key(&SECP256K1, &sk);
+
+                        let hash = hash160::Hash::hash(&pk.serialize());
+                        let script = Builder::new()
+                            .push_opcode(opcodes::all::OP_DUP)
+                            .push_opcode(opcodes::all::OP_HASH160)
+                            .push_slice(&hash.into_inner())
+                            .push_opcode(opcodes::all::OP_EQUALVERIFY)
+                            .push_opcode(opcodes::all::OP_CHECKSIG)
+                            .into_script();
+
+                        let sighash = SigHashCache::new(&tx).segwitv0_sighash(
+                            index as usize,
+                            &script,
+                            value,
+                            SigHashType::All,
+                        );
+
+                        let sig = SECP256K1.sign(&Message::from(sighash), &sk);
+
+                        let mut serialized_signature = sig.serialize_der().to_vec();
+                        serialized_signature.push(SigHashType::All as u8);
+
+                        tx.input[index as usize].witness.script_witness =
+                            vec![serialized_signature, pk.serialize().to_vec()];
+
+                        Ok(tx)
+                    }
+                },
+                // TODO: Do the same as in the loan transaction for the fee
+                Amount::from_sat(10_000),
+            )
+            .await
+            .unwrap();
+
+        client
+            .send_raw_transaction(&loan_repayment_transaction)
+            .await
+            .unwrap();
+    }
+
+    fn init_logger() {
+        // force enabling log output
+        // TODO: remove me again
```

Comment:
:question: @bonomat So do you want to remove this helper function or keep it?

Reply:
Keep the `init_logger` function but remove the next line: `env::set_var("RUST_LOG", "DEBUG");` which should be set from the outside